packages/pilot/src/containers/Filter/index.js

```diff
@@ -77,246 +62,110 @@ export const stringifyDates = ifElse(
+
+  useEffect(() => {
+    setFilterQuery(query)
+  }, [query])
+
+  useEffect(() => {
+    const {
+      filters: localFilters,
+    } = filterQuery
 
-    const prevQuery = stringifyDates(prevState.query)
-    const currQuery = stringifyDates(query)
+    const {
+      filters: urlFilters,
+    } = query
 
-    if (!equals(prevQuery, currQuery)) {
-      this.setState({ // eslint-disable-line react/no-did-update-set-state
-        hasChanged: true,
-        query,
-      })
-    }
-  }
+    setSelectedFilters(
+      getFilters(urlFilters, localFilters)
+    )
+  }, [filterQuery, query])
 
-  handleToogeMoreFilters () {
-    this.setState(({ collapsed }) => ({
-      collapsed: !collapsed,
-    }))
+  const handleToogleMoreFilters = () => {
+    setCollapsed(!collapsed)
   }
 
-  handleFiltersSubmit (filters) {
-    const { onConfirm } = this.props
-
-    this.setState({
-      collapsed: true,
-      hasChanged: false,
-    })
-
+  const handleFiltersSubmit = (filters) => {
+    setCollapsed(true)
     onConfirm(filters)
   }
 
-  handleFiltersChange (query) {
-    const { onChange } = this.props
-
-    this.setState({ query })
-
-    onChange(query)
-  }
-
-  renderChildrenInput (input, index) {
-    const { disabled } = this.props
-
-    return React.cloneElement(input, {
-      className: style.search,
-      disabled,
-      key: `${input.props.name}-${index}`,
-    })
-  }
-
-  renderToolbar () {
-    const {
-      children,
-      clearFilterDisabled,
-      confirmationDisabled,
-      disabled,
-      onClear,
-      options,
-      t,
-    } = this.props
-
-    const {
-      collapsed,
-      hasChanged,
-    } = this.state
-
-    return (
-      <CardActions>
-        {ensureArray(children).map(this.renderChildrenInput)}
-        {!isNilOrEmpty(options)
-          && (
-            <Button
-              disabled={disabled}
-              relevance="low"
-              fill="outline"
-              iconAlignment="end"
-              icon={collapsed
-                ? <ChevronDown32 width={16} height={16} />
-                : <ChevronUp32 width={16} height={16} />
-              }
-              onClick={this.handleToogeMoreFilters}
-            >
-              {t('components.filter.more')}
-            </Button>
-          )
-        }
-        <Button
-          relevance={
-            hasChanged
-              ? 'normal'
-              : 'low'
-          }
-          onClick={onClear}
-          fill="outline"
-          disabled={clearFilterDisabled || disabled}
-        >
-          {t('components.filter.reset')}
-        </Button>
-
-        <Button
-          relevance={
-            hasChanged
-              ? 'normal'
-              : 'low'
-          }
-          disabled={confirmationDisabled || !hasChanged || disabled}
-          type="submit"
-          fill="gradient"
-        >
-          {t('components.filter.apply')}
-        </Button>
-      </CardActions>
-    )
+  const handleFiltersChange = (NewQuery) => {
```

Comment:
esse 'n' pode ser minúsculo

```suggestion
  const handleFiltersChange = (newQuery) => {
```

Reply:
Boa, corrigido